src/Plugin.Maui.CalendarStore/CalendarStore.android.cs

```diff
@@ -377,12 +389,252 @@ public async Task UpdateEvent(string eventId, string title, string description,
+
+		var updateCount = platformContentResolver?.Update(
+			ContentUris.WithAppendedId(eventsTableUri, platformEventId), eventToUpdate, null, null);
+
+		if (updateCount != 1)
+		{
+			throw new CalendarStoreException($"Failed to update event with ID {eventId}.");
+		}
+
+		UpdateOrAddReminder(platformEventId, reminderMinutes);
+	}
+
+	void UpdateOrAddReminder(long eventId, int reminderMinutes)
+	{
+		var reminderSelection = $"{CalendarContract.Reminders.InterfaceConsts.EventId} = ?";
+		var reminderSelectionArgs = new[] { eventId.ToString() };
+
+		platformContentResolver?.Delete(CalendarContract.Reminders.ContentUri, reminderSelection, reminderSelectionArgs);
 	}
 
 	/// <inheritdoc/>
 	public Task UpdateEvent(CalendarEvent eventToUpdate) =>
 		UpdateEvent(eventToUpdate.Id, eventToUpdate.Title, eventToUpdate.Description,
-			eventToUpdate.Location, eventToUpdate.StartDate, eventToUpdate.EndDate, eventToUpdate.IsAllDay);
+			eventToUpdate.Location, eventToUpdate.StartDate, eventToUpdate.EndDate, eventToUpdate.IsAllDay, eventToUpdate.Reminders.ToArray());
+
+	/// <inheritdoc/>
+	void AddMultipleReminders(long eventId, Reminder[]? reminders)
+	{
+		if (reminders is null && reminders?.Length < 1)
+		{
+			return; //nothing to be done.
+		}
+
+		var existingReminders = GetExistingReminders(eventId);
+
+		int index = 0;
+
+		foreach (var reminder in reminders)
+		{
+			// Add a new reminder
+			AddSingleReminder(eventId, reminder.ReminderInMinutes);
+		}
+
+		// Remove any extra existing reminders not present in the new list
+		for (int i = index; i < existingReminders.Count; i++)
+		{
+			RemoveSingleReminder(existingReminders[i]);
+		}
+	}
+
+	public void RemoveSingleReminder(long eventId)
+	{
+		var reminderSelection = $"{CalendarContract.Reminders.InterfaceConsts.EventId} = ?";
+		var reminderSelectionArgs = new[] { eventId.ToString() };
+
+		var rowsDeleted = platformContentResolver?.Delete(
+			CalendarContract.Reminders.ContentUri, reminderSelection, reminderSelectionArgs);
+
+		if (rowsDeleted == null || rowsDeleted < 1)
+		{
+			// No reminder found, no need to throw error.
+			// Simply nothing was removed.
+			return;
+		}
+
+		//if it returns here, then reminder was removed.
+		//if you need a confirmation of sorts, make it return int for count or so
+	}
+	void RemoveAllReminders(long eventId)
+	{
+		var selection = $"{CalendarContract.Reminders.InterfaceConsts.EventId} = ?";
+		var selectionArgs = new[] { eventId.ToString() };
+
+		var rowsDeleted = platformContentResolver?.Delete(
+			CalendarContract.Reminders.ContentUri,
+			selection,
+			selectionArgs);
+
+		if (rowsDeleted == null || rowsDeleted <= 0)
+		{
+			Console.WriteLine("No reminders were found to delete, or an error occurred.");
+		}
+		else
+		{
+			Console.WriteLine($"Deleted {rowsDeleted} reminder(s) for event ID {eventId}.");
+		}
+	}
+
+	void AddSingleReminder(long eventId, int reminderMinutes)
```

Comment:
I don't see why we need separate calls for single and all/multiple reminders? Probably has to do with that minutes to reminder field. Just remove that and let people add reminders through the `Reminder` object. 

We can think about maybe making easier helper calls that will just add 1 reminder by specifying some minutes in a follow up PR.

Reply:
Removed AddSingleReminder.

Future PRs may indeed add more.